_analyzers/character-filters/pattern-replace-character-filter.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+title: Pattern Replace Character Filter
+parent: Character Filters
+nav_order: 130
+---
+
+# Pattern Replace Character Filter
+
+The `pattern replace` character filter allows you to use regular expressions (regex) to define patterns for matching and replacing characters in the input text. It is a flexible tool for advanced text transformations, especially when dealing with complex string patterns.
+
+This filter replaces all instances of a pattern with a specified replacement string, allowing for easy substitutions, deletions, or complex modifications to the input text. You can use it to normalize the input before tokenization.
+
+## Example of the pattern replacement character filter
+
+The following example standardizes phone numbers by removing spaces, dashes, or parentheses. 
+
+```json
+GET /_analyze
+{
+  "tokenizer": "standard",
+  "char_filter": [
+    {
+      "type": "pattern_replace",
+      "pattern": "[\\s()-]+",
+      "replacement": ""
+    }
+  ],
+  "text": "Call me at (555) 123-4567 or 555 987 6543"
+}
+```
+
+Running this `pattern replace` filter to remove spaces, dashes, and parentheses by replacing the pattern `[\\s()-]+` with an empty string will result in the following text:
+
+```
+Call me at 5551234567 or 5559876543 
```

Comment:
This won't be the out put as we are relacing all the whitespaces with ""

Correct output would be concatenated string

```
Callmeat5551234567or5559876543
```